utils/controller.js

```diff
@@ -0,0 +1,296 @@
+                user.stockShorted.id(companys._id).TotalStock += req.body.NoOfShares;
+            }
+            else{
+                user.stockShorted.push({_id : companys._id,TotalPrice : req.body.NoOfShares*companys.sharePrice ,TotalStock: req.body.NoOfShares});
+            }
+        }
+        companys.save();
+        user.save();
+        return res.json({success :"true",Company : companys,Customer : user});
+    })
+},
+coverstock(req,res){
+
+    let user ;
+    User.findById(req.body.id,(err,data)=>{
+        if(err) throw err;
+        user = data;})
+    
+    Company.findById(req.body.companyId,(err,companys)=>{
+        if(err) res.json(err);
+        if(!user.stockShorted.id(companys._id))
+        {
+            return res.json({msg :"shorted stock nhi he"});
+        }
+        if(user.stockShorted.id(companys._id).TotalStock<req.body.NoOfShares){
+           return res.json({msg :"itne shorted stock nhi he"});
+        }
+        else{
+            var historytemp  = {
+                sharePrice : companys.sharePrice,
+                availableQuantity : companys.availableQuantity-req.body.NoOfShares
+            }
+            var activitytemp ={
+                company : companys._id,
+                action : "covered" ,
+    	        quantity : req.body.NoOfShares,
+    	        price : companys.sharePrice
+            }
+            companys.history.push(historytemp);
+            let temp = user.stockShorted.id(companys._id)
+            user.accountBalance += Math.round(((temp.TotalPrice*req.body.NoOfShares)/temp.TotalStock)-companys.sharePrice*req.body.NoOfShares);
+            companys.availableQuantity -= req.body.NoOfShares;
+            user.activity.push(activitytemp);
+            if(user.stockShorted.id(companys._id))
+            {
+                user.stockShorted.id(companys._id).TotalPrice = Math.round(temp.TotalPrice*(1-(req.body.NoOfShares/temp.TotalStock)));
+                user.stockShorted.id(companys._id).TotalStock -= req.body.NoOfShares;
+             
+            }
+            
+        }
+        companys.save();
+        user.save();
+        return res.json({success :"true",Company : companys,Customer : user});
+    })
+
+},
+takeloan(req,res)
+{
+    let user ;
+    User.findById(req.body.id,(err,data)=>{
+        if(err) throw err;
+        user = data;
+        if(user.loan[0].amount+req.body.loan>parameter.maxLoan)
+        {
+            return res.json({msg:"aur nhi "});
+        }
+        user.loan[0].isPending = true;
+        user.loan[0].amount += req.body.loan;
+        user.accountBalance += req.body.loan;
+        user.save();
+        res.json({
+            Customer : data
+        })
+    })
+
+         
+},
+repayloan(req,res)
+{
+    let user ;
+    User.findById(req.body.id,(err,data)=>{
+        if(err) throw err;
+        user = data;
+        if(user.loan[0].amount-req.body.loan<0 )
```

Comment:
dont make loan into an array. We just need to save whether the loan is pending, and the amount of loan given. We dont require the entire history. We'll add/subtract loan in the total worth just by looking at whether the loan is pending or not

Reply:
ok will do so